src/chains/filecoin/filecoin/src/deal-state.ts

```diff
@@ -1,46 +1,127 @@
 // Note that the runtime number values of these enums match
-// *exactly* the number values of the same states on the Lotus server.
+// *exactly* the number values of the same states found at
+// go-fil-markets/storagemarket/dealstatus.go (https://git.io/JtJAS)
+// **DO NOT TRUST** other sources (i.e. lotus/lotuspond/front/src/Client.js
+// or js-lotus-client-workshop/src/08-deals/deal-list.js)
 // Don't reorganize unless you know what you're doing.
```

Comment:
Since this is less of an `enum` and more of a mapping, should we just convert this to a map?

Or at least explicitly type the integer values instead of relying on position? .e.g,

```
export enum DealState {
  // Unknown means the current status of a deal is undefined
  Unknown = 0,
  ```
  
This is just a question, not a request.

Reply:
> Since this is less of an enum and more of a mapping, should we just convert this to a map?

This definitely is an enum rather than a map; I personally can't think of why this would be considered a map. It's implemented in golang as an ordered enum and it's logical purpose is an enum.

> Or at least explicitly type the integer values instead of relying on position? .e.g,

The thing is, that this code is essentially a direct port from the golang implementation. They don't use numbers in their implementation, so I favored not using them here to be consistent with their implementation.